pkg/plugin/plugin.go

```diff
@@ -112,62 +147,101 @@ type Plugin struct {
 	Dir string
 }
 
-// The following rules will apply to processing the Plugin.PlatformCommand.Command:
-// - If both OS and Arch match the current platform, search will stop and the command will be prepared for execution
-// - If OS matches and there is no more specific match, the command will be prepared for execution
-// - If no OS/Arch match is found, return nil
-func getPlatformCommand(cmds []PlatformCommand) []string {
-	var command []string
+// Returns command and args strings based on the following rules in priority order:
+// - From the PlatformCommand where OS and Arch match the current platform
+// - From the PlatformCommand where OS matches the current platform and Arch is empty
+// - From the PlatformCommand where OS and Arch are both empty
+// - Return nil, nil
+func getPlatformCommand(cmds []PlatformCommand) ([]string, []string) {
+	var command, args []string
+	found := false
+
 	eq := strings.EqualFold
 	for _, c := range cmds {
-		if eq(c.OperatingSystem, runtime.GOOS) {
-			command = strings.Split(c.Command, " ")
-		}
 		if eq(c.OperatingSystem, runtime.GOOS) && eq(c.Architecture, runtime.GOARCH) {
-			return strings.Split(c.Command, " ")
+			return strings.Split(c.Command, " "), c.Args
+		}
+
+		if len(c.Architecture) > 0 && !eq(c.Architecture, runtime.GOARCH) {
+			continue
+		}
+
+		if eq(c.OperatingSystem, runtime.GOOS) || (len(c.OperatingSystem) == 0 && !found) {
```

Comment:
I think it would be simpler to iterate a third time and do:

```
	for _, c := range cmds {
	 	 if eq(c.OperatingSystem, runtime.GOOS) && c.Architecture == "" {
	 		 return strings.Split(c.Command, " "), c.Args
	 	}
	 }
	 
	for _, c := range cmds {
	 	 if c.OperatingSystem == "" and c.Architecture == "" {
	 		 return strings.Split(c.Command, " "), c.Args
	 	}
	 }
	 
	 return nil, nil
```

rather than overload this loop with the boolean `found`

Reply:
I've updated the function for the above comment, but it still uses a single loop. To work correctly with multiple loops we'd need 4 loops (see below) so I think the more complex pattern is valid.

- Exact match
- OS match + empty arch
- Arch match + empty OS
- Empty OS + empty arch

Reply:
👍

I'm not sure if "Arch match + empty OS" makes sense? I don't know of binaries built for a specific arch, that run over all OSs? But support doesn't hurt (aside from the additional, already written, logic)

Reply:
(and I could be wrong)

Reply:
I didn't include it in the first place but I added it for completeness. But in this case I'd be happy to lower the complexity and make arch only valid when OS was also set. WDYT?

Reply:
I need to make another change so I'll remove the arch only matching logic at the same time.